File: appium-tests/src/test/kotlin/pizza/evan/pages/HelloWorldPage.kt
package pizza.evan.pages

import io.appium.java_client.AppiumDriver
import io.appium.java_client.pagefactory.AndroidFindBy
import io.appium.java_client.pagefactory.AppiumFieldDecorator
import io.appium.java_client.pagefactory.iOSXCUITFindBy
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.PageFactory
import pizza.evan.appium.accessibilityString
import pizza.evan.appium.string

class HelloWorldPage(
    driver: AppiumDriver
): Page {

    @iOSXCUITFindBy(accessibility = "greeting")
    @AndroidFindBy(id = "greeting")
    private lateinit var greetingLabel: WebElement

    @iOSXCUITFindBy(accessibility = "icon")
    @AndroidFindBy(id = "icon")
    private lateinit var iconImage: WebElement

    @iOSXCUITFindBy(accessibility = "next_button")
    @AndroidFindBy(id = "next_button")
    private lateinit var nextButton: WebElement

    init {
        PageFactory.initElements(AppiumFieldDecorator(driver), this)
    }

    val greetingText: String
        get() = greetingLabel.string

    val iconAccessibilityText: String
        get() = iconImage.accessibilityString

    override fun close() { /* root page */ }

    fun tapNextButton() {
        nextButton.click()
    }
}

File: iOS/UI Tests/Pages/HelloWorldPage.swift
//
//  Copyright © 2024 Evan DeLaney. All rights reserved.
//

import XCTest

final class HelloWorldPage {
    
    private let greetingLabel: XCUIElement
    
    private let iconImage: XCUIElement
    
    var greetingText: String {
        greetingLabel.label
    }
    
    var iconAccessibilityText: String {
        iconImage.label
    }
    
    init(app: XCUIApplication)
    {
        greetingLabel = app.staticTexts["greeting"]
        iconImage = app.images["icon"]
    }
}

